src/avatar_action.cpp

```diff
@@ -846,6 +849,88 @@ bool avatar_action::fire( avatar &you, map &m, item &weapon, int bp_cost )
     return avatar_action::fire( you, m );
 }
 
+bool avatar_action::eat_here( avatar &you )
+{
+    if( ( you.has_active_mutation( trait_RUMINANT ) || you.has_active_mutation( trait_GRAZER ) ) &&
+        ( g->m.ter( you.pos() ) == t_underbrush || g->m.ter( you.pos() ) == t_shrub ) ) {
+        if( you.get_hunger() < 20 ) {
+            add_msg( _( "You're too full to eat the leaves from the %s." ), g->m.ter( you.pos() )->name() );
+            return true;
+        } else {
+            you.moves -= 400;
+            g->m.ter_set( you.pos(), t_grass );
+            add_msg( _( "You eat the underbrush." ) );
+            item food( "underbrush", calendar::turn, 1 );
+            you.eat( food );
+            return true;
+        }
+    }
+    if( you.has_active_mutation( trait_GRAZER ) && ( g->m.ter( you.pos() ) == t_grass ||
+            g->m.ter( you.pos() ) == t_grass_long || g->m.ter( you.pos() ) == t_grass_tall ) ) {
+        if( you.get_hunger() < 8 ) {
+            add_msg( _( "You're too full to graze." ) );
+            return true;
+        } else {
+            you.moves -= 400;
+            add_msg( _( "You eat the grass." ) );
+            item food( item( "grass", calendar::turn, 1 ) );
+            you.eat( food );
+            g->m.ter_set( you.pos(), t_dirt );
+            if( g->m.ter( you.pos() ) == t_grass_tall ) {
```

Comment:
These checks here are a bit pointless. In the line above, the terrain was explicitly set to "t_dirt", that's what it will be here as well, so the checks never succeed and the falls through to the end where it sets the terrain to "t_dirt" *again*.

Reply:
good catch. apparently that's just like that in `master`. i'll take care of it, because obviously that can't be the desired functionality.